nano/node/fair_queue.hpp

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include <nano/lib/locks.hpp>
+#include <nano/lib/numbers.hpp>
+#include <nano/lib/stats.hpp>
+#include <nano/lib/thread_roles.hpp>
+#include <nano/lib/threading.hpp>
+#include <nano/lib/utility.hpp>
+#include <nano/node/bandwidth_limiter.hpp>
+#include <nano/node/transport/channel.hpp>
+
+#include <boost/circular_buffer.hpp>
+
+#include <algorithm>
+#include <chrono>
+#include <condition_variable>
+#include <deque>
+#include <functional>
+#include <mutex>
+#include <thread>
+#include <utility>
+#include <vector>
+
+namespace nano
+{
+template <typename Request, typename... Sources>
```

Comment:
I see this allows variadic type arguments but they aren't used. What do you envision this being used for? Could be the user's responsibility to use a std::tuple argument here if needed?

Reply:
I anticipate variadic arguments to be used for "multi-level" prioritization, ie. in case of vote processor I want to prioritize by a <vote source (live, cache), rep tier (tier 1, 2, 3, none)> tuple. So that each source and each rep tier can get different priority and queue size. This can be simplified to be responsibility of user to specify, yeah.